snapcraft/internal/pluginhandler.py

```diff
@@ -324,23 +324,9 @@ def migratable_fileset_for(self, step):
         return _migratable_filesets(fileset, self.code.installdir)
 
     def _organize(self):
-        organize_fileset = getattr(self.code.options, 'organize', {}) or {}
+        fileset = getattr(self.code.options, 'organize', {}) or {}
 
-        for key in organize_fileset:
-            src = os.path.join(self.code.installdir, key)
-            dst = os.path.join(self.code.installdir, organize_fileset[key])
-
-            os.makedirs(os.path.dirname(dst), exist_ok=True)
-
-            if os.path.exists(dst):
-                logger.warning(
-                    'Stepping over existing file for organization %r',
-                    os.path.relpath(dst, self.code.installdir))
-                if os.path.isdir(dst):
-                    shutil.rmtree(dst)
-                else:
-                    os.remove(dst)
-            shutil.move(src, dst)
+        _organize_filesets(fileset.copy(), self.code.installdir)
```

Comment:
But you don't rely on `fileset` being immutable here, do you? You don't use it again, the function exits here.